File: templates/components/vectordbs/typescript/llamacloud/queryFilter.ts
import type { MetadataFilter, MetadataFilters } from "@llamaindex/cloud/api";

export function generateFilters(documentIds: string[]): MetadataFilters {
  // public documents don't have the "private" field or it's set to "false"
  // @ts-ignore TODO: fix typing in @llamaindex/cloud/api
  const publicDocumentsFilter: MetadataFilter = {
    key: "private",
    operator: "is_empty",
  };

  // if no documentIds are provided, only retrieve information from public documents
  if (!documentIds.length) return { filters: [publicDocumentsFilter] };

  const privateDocumentsFilter: MetadataFilter = {
    key: "file_id", // Note: LLamaCloud uses "file_id" to reference private document ids as "doc_id" is a restricted field in LlamaCloud
    value: documentIds,
    operator: "in",
  };

  // if documentIds are provided, retrieve information from public and private documents
  return {
    filters: [publicDocumentsFilter, privateDocumentsFilter],
    condition: "or",
  };
}

File: templates/components/vectordbs/python/llamacloud/query_filter.py
from llama_index.core.vector_stores.types import MetadataFilter, MetadataFilters


def generate_filters(doc_ids):
    """
    Generate public/private document filters based on the doc_ids and the vector store.
    """
    # public documents (ingested by "poetry run generate" or in the LlamaCloud UI) don't have the "private" field
    public_doc_filter = MetadataFilter(
        key="private",
        value=None,
        operator="is_empty",  # type: ignore
    )
    selected_doc_filter = MetadataFilter(
        key="file_id",  # Note: LLamaCloud uses "file_id" to reference private document ids as "doc_id" is a restricted field in LlamaCloud
        value=doc_ids,
        operator="in",  # type: ignore
    )
    if len(doc_ids) > 0:
        # If doc_ids are provided, we will select both public and selected documents
        filters = MetadataFilters(
            filters=[
                public_doc_filter,
                selected_doc_filter,
            ],
            condition="or",  # type: ignore
        )
    else:
        filters = MetadataFilters(
            filters=[
                public_doc_filter,
            ]
        )

    return filters
